koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -856,23 +866,50 @@ def populate_node_table(self, cluster, nodes):
 
     def populate_pvc_table(self, cluster, pvcs):
         """Get or create an entry in the OCP cluster table."""
-        LOG.info(log_json(msg="populating reporting_ocp_pvcs table", schema=self.schema, cluster=cluster))
+
+        LOG.info(
+            log_json(
+                msg="populating reporting_ocp_pvcs table",
+                schema=self.schema,
+                cluster_id=cluster.cluster_id,
+                cluster_alias=cluster.cluster_alias,
+            )
+        )
+
         for pvc in pvcs:
-            try:
-                ocppvc = OCPPVC.objects.get(persistent_volume=pvc[0], persistent_volume_claim=pvc[1], cluster=cluster)
+            ocppvc = OCPPVC.objects.filter(
+                persistent_volume=pvc[0], persistent_volume_claim=pvc[1], cluster=cluster
+            ).first()
+            if ocppvc:
                 if not ocppvc.csi_volume_handle:
                     # Update the existing record's csi_volume_handle
                     ocppvc.csi_volume_handle = pvc[2]
                     ocppvc.save(update_fields=["csi_volume_handle"])
-            except OCPPVC.DoesNotExist:
-                # If the record does not exist, create a new one
-                OCPPVC.objects.create(
-                    persistent_volume=pvc[0], persistent_volume_claim=pvc[1], csi_volume_handle=pvc[2], cluster=cluster
-                )
+            else:
+                # If the record does not exist, try creating a new one
+                try:
+                    OCPPVC.objects.create(
+                        persistent_volume=pvc[0],
+                        persistent_volume_claim=pvc[1],
+                        csi_volume_handle=pvc[2],
+                        cluster=cluster,
+                    )
+
+                except IntegrityError:
+                    LOG.warning(f"IntegrityError raised when creating pvc: {pvc}.")
```

Comment:
It would still be good to log the exception:
```suggestion
                except IntegrityError as e:
                    LOG.warning(f"IntegrityError raised when creating pvc: {pvc}: {e}")
```

This line also needs a unit test.

Reply:
cleaned up the log and added a unit test.